### PR TITLE
Consolidate parameters

### PR DESCRIPTION
**Pull request by kmohrman**
This branch addresses [Issue #56](https://github.com/TopEFT/topcoffea/issues/56) by moving the values for the object selection cuts into a json file `topcoffea/json/params.json`. The object selection script and the processor access the values from the file using a function called [get_cut()](https://github.com/TopEFT/topcoffea/blob/consolidate-parameters/topcoffea/modules/GetValuesFromJsons.py#L13) added in a new file `topcoffea/modules/GetValuesFromJsons.py`. There is probably a better way to access the values, so please let me know of any suggestions. Otherwise, this seems to work okay for now.

I've checked the yields against the master branch (for UL16, UL17, and UL18 samples). The largest percent differences are on the order of 1e-16, so I don't think this change has introduced any new bugs (hopefully). 

This update should hopefully make it easier to keep track of the cuts we are using. It would also make it easier if we ever want to change any the cuts, since we'd only have to change them in one place.

### CODE REVIEW

File: analysis/topEFT/topeft.py [hunk 1]
```diff
@@ -160,20 +161,35 @@ def process(self, events):
...
-        # to do: check these numbers are ok
-        if year == "2017": btagwpl = 0.0532 #WP loose 
-        else: btagwpl = 0.0490 #WP loose 
+        # Loose DeepJet WP
+        # TODO: Update these numbers when UL16 is available, and double check UL17 and UL18 at that time as well
+        if year == "2017":
+            btagwpl = get_cut("btag_wp_loose_UL17")
+        elif year == "2018":
+            btagwpl = get_cut("btag_wp_loose_UL18")
+        elif ((year=="2016") or (year=="2016APV")):
```
Comment (bryates) [inline analysis/topEFT/topeft.py:175]: This is fine, but `elif "2016" in year:` also works.

Comment (kmohrman) [reply analysis/topEFT/topeft.py:175]: Thanks, yep, looks like that would work too. But I suppose that would not catch a typo where the year string includes `2016` (e.g. `20167` would just silently get mapped to 2016, even if you intended it to be `2017` and just accidentally typed a `6` in there for some reason), so I guess for that reason I'd probably prefer to keep checking for exact matches.

Comment (bryates) [reply analysis/topEFT/topeft.py:175]: That's a really good point!

**Comment by bryates** [comment]
Runs fine for me. I have no objections, other than an issue with the `check_yields.sh` script not working correctly, but this isn't isolated to this branch.

**Comment by bryates** [comment]
Results from the fixed `check_yields.sh` script. 
```
--- Percent diff between Ref yields and New yields ---

ttH
    cat_2lss_p                                                                       0.0025623503029261006 -> NOTE: This is larger than tolerance (1e-08)!    cat_2lss_m                                                                       -0.012248259164838024 -> NOTE: This is larger than tolerance (1e-08)!    cat_3l_onZ_1b
        -0.2612608952732432 -> NOTE: This is larger than tolerance (1e-08)!
    cat_3l_p_offZ_1b                                                                 -0.5293640998556665 -> NOTE: This is larger than tolerance (1e-08)!
    cat_3l_m_offZ_1b
        -0.4593797391774939 -> NOTE: This is larger than tolerance (1e-08)!
    cat_3l_onZ_2b
        -0.22024716273016648 -> NOTE: This is larger than tolerance (1e-08)!
    cat_3l_p_offZ_2b
        -0.41324479372995704 -> NOTE: This is larger than tolerance (1e-08)!
    cat_3l_m_offZ_2b                                                                 -0.46558439909668214 -> NOTE: This is larger than tolerance (1e-08)!
    cat_4l
        -0.2914898985847001 -> NOTE: This is larger than tolerance (1e-08)!
```

**Comment by kmohrman** [comment]
Thanks @bryates. The reason why the `comp_yields.py` script was upset was because the placeholder reference json was produced before we merged the `EventSelection` branch, and I had changed the naming conventions for some of the categories in `EventSelection`, so the keys in the new dictionary did not match the old dictionary, and the code did not know which values to compare, so it was not happy.

The reason why you did not see agreement after just changing the names of the categories is because the numbers in the placeholder reference json were from before we merged the `EventSelection` branch, so are very out of date. I will remake the reference json file and push it.

However, I should point out that I wrote the `check_yields.sh` script as more of a placeholder/example prototype (for what we eventually want to do with CI) than something I actually use. It's currently not even comparing an EFT sample. However, I think you make a good point while we're waiting for the CI to come into place, there's really no reason why we can't update the `check_yields.sh` script to do a more useful check. That way, we can use the `check_yields.sh` script instead of running the `get_yield_json.py` and `comp_yields.py` scripts by hand every time.

I'll plan to make those updates to `check_yields.sh` today, and once I push the changes please let me know if it matches what you had in mind.

**Comment by kmohrman** [comment]
@bryates, I've updated the check yields script to process EFT samples instead of central samples, to use the wq run script since this set of samples would take too long with futures, and to compare against up-to-date reference yields.

**Comment by bryates** [comment]
> @bryates, I've updated the check yields script to process EFT samples instead of central samples, to use the wq run script since this set of samples would take too long with futures, and to compare against up-to-date reference yields.

Looks great, thanks @kmohrman! I'm running it now.

**Comment by bryates** [comment]
Here's the updated check:
```
Comparing yields agains reference...
WARNING, key tHq is not in both dictionaries. Continuing...
WARNING, key tllq is not in both dictionaries. Continuing...
WARNING, key ttll is not in both dictionaries. Continuing...
WARNING, key ttlnu is not in both dictionaries. Continuing...
WARNING, key tttt is not in both dictionaries. Continuing...
WARNING, key tHq is not in both dictionaries. Continuing...
WARNING, key tllq is not in both dictionaries. Continuing...
WARNING, key ttll is not in both dictionaries. Continuing...
WARNING, key ttlnu is not in both dictionaries. Continuing...
WARNING, key tttt is not in both dictionaries. Continuing...

--- Percent diff between Ref yields and New yields ---

ttH
    cat_2lss_p
        0.049947118778003716 -> NOTE: This is larger than tolerance (1e-08)!
    cat_2lss_m
        0.03462020987432899 -> NOTE: This is larger than tolerance (1e-08)!
    cat_3l_p_offZ_1b
        0.04132654630319621 -> NOTE: This is larger than tolerance (1e-08)!
    cat_3l_m_offZ_1b
        -0.05452583090528013 -> NOTE: This is larger than tolerance (1e-08)!
    cat_3l_p_offZ_2b
        0.049361093845095676 -> NOTE: This is larger than tolerance (1e-08)!
    cat_3l_m_offZ_2b
        0.009878110442970593 -> NOTE: This is larger than tolerance (1e-08)!
    cat_3l_onZ_1b
        -0.0730850950759218 -> NOTE: This is larger than tolerance (1e-08)!
    cat_3l_onZ_2b
        -0.006323340477013478 -> NOTE: This is larger than tolerance (1e-08)!    cat_4l
        -0.02055848237482613 -> NOTE: This is larger than tolerance (1e-08)!
```

**Comment by kmohrman** [comment]
Hmm, the results for me were the following:
```
Comparing yields agains reference...

--- Percent diff between Ref yields and New yields ---

tHq
    cat_2lss_p
	-1.7143940957639726e-16
    cat_2lss_m
	0.0
    cat_3l_p_offZ_1b
	-1.678962665419917e-16
    cat_3l_m_offZ_1b
	-1.7367792331491767e-16
    cat_3l_p_offZ_2b
	-1.183075843823156e-16
    cat_3l_m_offZ_2b
	0.0
    cat_3l_onZ_1b
	0.0
    cat_3l_onZ_2b
	-3.0226661501282845e-16
    cat_4l
	0.0
tllq
    cat_2lss_p
	-3.779696169179378e-16
    cat_2lss_m
	-1.6266291748297503e-16
    cat_3l_p_offZ_1b
	-2.1514634280870112e-16
    cat_3l_m_offZ_1b
	-1.9507249038439e-16
    cat_3l_p_offZ_2b
	-2.493571518422034e-16
    cat_3l_m_offZ_2b
	-2.2061364863441973e-16
    cat_3l_onZ_1b
	-2.9613715370050695e-16
    cat_3l_onZ_2b
	-3.44426616648829e-16
    cat_4l
	-2.6727115069437493e-16
ttH
    cat_2lss_p
	-3.5158517498849307e-16
    cat_2lss_m
	-3.513148795880848e-16
    cat_3l_p_offZ_1b
	-3.704265485510273e-16
    cat_3l_m_offZ_1b
	-3.7878058935876545e-16
    cat_3l_p_offZ_2b
	-3.8325852706538557e-16
    cat_3l_m_offZ_2b
	-3.9534593856409334e-16
    cat_3l_onZ_1b
	-4.101558107382045e-16
    cat_3l_onZ_2b
	-6.081049652824652e-16
    cat_4l
	-5.121788739791817e-16
ttll
    cat_2lss_p
	0.0
    cat_2lss_m
	1.892092661190597e-16
    cat_3l_p_offZ_1b
	4.259920858374154e-16
    cat_3l_m_offZ_1b
	2.2445623763765824e-16
    cat_3l_p_offZ_2b
	2.352703638429882e-16
    cat_3l_m_offZ_2b
	1.114119347314511e-16
    cat_3l_onZ_1b
	0.0
    cat_3l_onZ_2b
	2.467125060254119e-16
    cat_4l
	1.2999857441002981e-16
ttlnu
    cat_2lss_p
	-1.2372677824881061e-16
    cat_2lss_m
	-1.1660353694861883e-16
    cat_3l_p_offZ_1b
	-2.1778174747492955e-16
    cat_3l_m_offZ_1b
	-2.0741504358310909e-16
    cat_3l_p_offZ_2b
	-2.0650179257318295e-16
    cat_3l_m_offZ_2b
	0.0
    cat_3l_onZ_1b
	-1.1188208617434003e-16
    cat_3l_onZ_2b
	-2.1734820050066655e-16
    cat_4l
	-1.4537763302562738e-16
tttt
    cat_2lss_p
	-5.585890905039128e-16
    cat_2lss_m
	-4.121442306891474e-16
    cat_3l_p_offZ_1b
	-6.23755624174993e-16
    cat_3l_m_offZ_1b
	-3.914425746874236e-16
    cat_3l_p_offZ_2b
	-5.858540899726749e-16
    cat_3l_m_offZ_2b
	-5.943771134553073e-16
    cat_3l_onZ_1b
	-5.728547842990262e-16
    cat_3l_onZ_2b
	-5.081639708319029e-16
    cat_4l
	-3.679821866959206e-16
```

**Comment by bryates** [comment]
> Hmm, the results for me were the following:
> 
> ```
> Comparing yields agains reference...
> 
> --- Percent diff between Ref yields and New yields ---
> 
> tHq
>     cat_2lss_p
> 	-1.7143940957639726e-16
>     cat_2lss_m
> 	0.0
>     cat_3l_p_offZ_1b
> 	-1.678962665419917e-16
>     cat_3l_m_offZ_1b
> 	-1.7367792331491767e-16
>     cat_3l_p_offZ_2b
> 	-1.183075843823156e-16
>     cat_3l_m_offZ_2b
> 	0.0
>     cat_3l_onZ_1b
> 	0.0
>     cat_3l_onZ_2b
> 	-3.0226661501282845e-16
>     cat_4l
> 	0.0
> tllq
>     cat_2lss_p
> 	-3.779696169179378e-16
>     cat_2lss_m
> 	-1.6266291748297503e-16
>     cat_3l_p_offZ_1b
> 	-2.1514634280870112e-16
>     cat_3l_m_offZ_1b
> 	-1.9507249038439e-16
>     cat_3l_p_offZ_2b
> 	-2.493571518422034e-16
>     cat_3l_m_offZ_2b
> 	-2.2061364863441973e-16
>     cat_3l_onZ_1b
> 	-2.9613715370050695e-16
>     cat_3l_onZ_2b
> 	-3.44426616648829e-16
>     cat_4l
> 	-2.6727115069437493e-16
> ttH
>     cat_2lss_p
> 	-3.5158517498849307e-16
>     cat_2lss_m
> 	-3.513148795880848e-16
>     cat_3l_p_offZ_1b
> 	-3.704265485510273e-16
>     cat_3l_m_offZ_1b
> 	-3.7878058935876545e-16
>     cat_3l_p_offZ_2b
> 	-3.8325852706538557e-16
>     cat_3l_m_offZ_2b
> 	-3.9534593856409334e-16
>     cat_3l_onZ_1b
> 	-4.101558107382045e-16
>     cat_3l_onZ_2b
> 	-6.081049652824652e-16
>     cat_4l
> 	-5.121788739791817e-16
> ttll
>     cat_2lss_p
> 	0.0
>     cat_2lss_m
> 	1.892092661190597e-16
>     cat_3l_p_offZ_1b
> 	4.259920858374154e-16
>     cat_3l_m_offZ_1b
> 	2.2445623763765824e-16
>     cat_3l_p_offZ_2b
> 	2.352703638429882e-16
>     cat_3l_m_offZ_2b
> 	1.114119347314511e-16
>     cat_3l_onZ_1b
> 	0.0
>     cat_3l_onZ_2b
> 	2.467125060254119e-16
>     cat_4l
> 	1.2999857441002981e-16
> ttlnu
>     cat_2lss_p
> 	-1.2372677824881061e-16
>     cat_2lss_m
> 	-1.1660353694861883e-16
>     cat_3l_p_offZ_1b
> 	-2.1778174747492955e-16
>     cat_3l_m_offZ_1b
> 	-2.0741504358310909e-16
>     cat_3l_p_offZ_2b
> 	-2.0650179257318295e-16
>     cat_3l_m_offZ_2b
> 	0.0
>     cat_3l_onZ_1b
> 	-1.1188208617434003e-16
>     cat_3l_onZ_2b
> 	-2.1734820050066655e-16
>     cat_4l
> 	-1.4537763302562738e-16
> tttt
>     cat_2lss_p
> 	-5.585890905039128e-16
>     cat_2lss_m
> 	-4.121442306891474e-16
>     cat_3l_p_offZ_1b
> 	-6.23755624174993e-16
>     cat_3l_m_offZ_1b
> 	-3.914425746874236e-16
>     cat_3l_p_offZ_2b
> 	-5.858540899726749e-16
>     cat_3l_m_offZ_2b
> 	-5.943771134553073e-16
>     cat_3l_onZ_1b
> 	-5.728547842990262e-16
>     cat_3l_onZ_2b
> 	-5.081639708319029e-16
>     cat_4l
> 	-3.679821866959206e-16
> ```

After reinitializing my grid proxy, I now get the same results! If there's nothing else left, I think we can merge this PR.

**Comment by kmohrman** [comment]
Great, thanks for confirming the results @bryates! Also, thanks for noticing that the placeholder reference yields were out of date, and that the check yields script was not doing any useful checks.

**Pull request merged**